xayn-ai/src/data/mod.rs

```diff
@@ -44,10 +69,21 @@ impl NegativeCoi {
             point,
         }
     }
+
+    pub fn merge(&self, other: &Self, id: usize) -> Self {
+        let point = mean(&self.point, &other.point);
+        Self {
+            id: CoiId(id),
+            point,
+        }
+    }
 }
 
 pub(crate) trait CoiPoint {
     fn new(id: usize, embedding: Embedding) -> Self;
+    fn merge(&self, other: &Self, id: usize) -> Self;
```

Comment:
Can we move `self` and `other` instead of taking them by reference? I think it reflect better the intent of the merge.